util/topgen/merge.py

```diff
@@ -173,6 +173,11 @@ def elaborate_instance(instance, block: IpBlock):
         if isinstance(s['width'], Parameter):
             for p in instance["param_list"]:
                 if p['name'] == s['width'].name:
+                    # When mangeling the name, we first need to deep copy the param. Otherwise, we
+                    # reference always the same single parameter. If there are multiple instances
+                    # having the same parameter, we would overwrite the previously mangled param
```

Comment:
I think this happens if there are multiple signals whose width is given by the same parameter? Could the text be expanded slightly? (In particular, I didn't immediately know what "having the same parameter" meant)

Reply:
Oh, I see. Yep, that makes sense (and thanks for explaining it!)